### ISSUE TITLE
css/WOFF2 tests overlap with woff2-compiled-tests repo

### ISSUE THREAD

**Issue by foolip** [w3c/web-platform-tests#9300]
These are very similar:
https://github.com/w3c/web-platform-tests/tree/master/css/WOFF2
https://github.com/w3c/woff2-compiled-tests/tree/master/UserAgent/Tests/xhtml1

Someone built from https://github.com/w3c/woff2-tests/blob/master/generators/UserAgentTestCaseGenerator.py

People who have poked at the woff2-compiled-tests repo are @rsheeter and @khaledhosny.

Looks like @svgeesus imported the tests into csswg-test, which has now been merged into this repo, but I can't tell what the origin is.

It would be good if there were one source of truth, and if any divergence between these is resolved.

The situation is a bit similar to WebAssembly I think, and @jgraham might have opinions on how this should work.

**Comment by foolip** [comment]
Hmm, the times of these commits give a clue:
https://github.com/w3c/web-platform-tests/commit/5e10afdc69111c73be880f2048093aeb3fb48187
https://github.com/w3c/woff2-tests/commit/64c51c131040de1bf098c93723f57c2861a1bfde

Seems like the tests must have existed in a third place that both were sourced from?

**Comment by khaledhosny** [comment]
https://github.com/w3c/woff2-tests contain the scripts that generates the tests, https://github.com/w3c/woff2-compiled-tests contains the output of running these scripts.

**Comment by foolip** [comment]
@khaledhosny what was the source of the templates in UserAgentTestCaseGenerator.py in https://github.com/w3c/woff2-tests/commit/64c51c131040de1bf098c93723f57c2861a1bfde?

**Comment by khaledhosny** [comment]
The WOFF test suite https://dev.w3.org/webfonts/WOFF/tests/.

**Comment by foolip** [comment]
@sideshowbarker, would you happen to know if https://dev.w3.org/webfonts/WOFF/tests/ is still maintained, or what we should now currently consider the canonical source of these tests?

**Comment by khaledhosny** [comment]
No idea.

**Comment by sideshowbarker** [comment]
> @sideshowbarker, would you happen to know if https://dev.w3.org/webfonts/WOFF/tests/ is still maintained, or what we should now currently consider the canonical source of these tests?

I reckon @svgeesus would know

**Comment by fred-wang** [comment]
It would probably be nice to move both the source and output into the web-platform-tests repository (see https://github.com/w3c/web-platform-tests#adding-command-line-scripts-tools-subdirs for tools/ directories).

I think we should also consider using only web fonts as discussed in #8440 and https://github.com/w3c/web-platform-tests/pull/8494 otherwise the tests don't work in browsers without manual installation.

**Comment by svgeesus** [comment]
> Looks like @svgeesus imported the tests into csswg-test, which has now been merged into this repo,

Correct

> but I can't tell what the origin is.

The origin was https://github.com/w3c/woff2-tests which in turn were reviewed by the WebFonts WG. That is the canonical source (but they need to be built, they can't be used as-is).

> I think we should also consider using only web fonts as discussed in #8440 and #8494 otherwise the tests don't work in browsers without manual installation.

That gives you a bootstrapping problem, you are testing webfonts and relying on webfont support to do so.

**Comment by fred-wang** [comment]
> That gives you a bootstrapping problem, you are testing webfonts and relying on webfont support to do so.

@svgeesus The idea is to rely on the now widely supported WOFF1 format to test the WOFF2 format as you commented in https://github.com/w3c/web-platform-tests/issues/8440#issuecomment-351769574

**Comment by svgeesus** [comment]
@fred-wang ah okay, fair enough.

**Comment by fred-wang** [comment]
Hi, I just sent two PRs to make the the WPT source and the woff2-tests output closer:
https://github.com/web-platform-tests/wpt/pull/11311
https://github.com/w3c/woff2-tests/pull/10

**Comment by foolip** [comment]
Cool, thanks!

**Comment by fred-wang** [comment]
Two more PRs:
https://github.com/web-platform-tests/wpt/pull/11313
https://github.com/web-platform-tests/wpt/pull/11314

After that the only remaining differences with the output of https://github.com/w3c/woff2-tests will be:
* The available-00* files (I'm still not sure what to do with these tests, see also https://github.com/web-platform-tests/wpt/issues/8482 )
* Maybe some binary font files that have changed since the import.
* The OWNERS files is not generated by woff2-tests (do we want it?)

Also, I think we should at least make woff2-tests add some warning in the output like "THIS FILE WAS AUTOMATICALLY GENERATED, DO NOT EDIT".

WDYT?

**Comment by fred-wang** [comment]
https://github.com/web-platform-tests/wpt/tree/master/css/WOFF2 and https://github.com/w3c/woff2-tests/blob/master/generators/UserAgentTestCaseGenerator.py are now in sync (except that OWNERS files is not generated). I did not modify binary fonts, maybe @khaledhosny can confirm they are up-to-date. Finally, I also added the DO NOT EDIT warning.

@foolip Do you think anything else should be done here besides maybe https://github.com/w3c/woff2-compiled-tests/pull/1 ?

**Comment by foolip** [comment]
@fred-wang, nope, looks like you've very thoroughly fixed this issue, thank you!

Do you want to sync the binary fonts as well before closing this issue?

**Comment by fred-wang** [comment]
> Do you want to sync the binary fonts as well before closing this issue?

Probably we could carefully check the log history or the diff of ttx output but let's just blindly force an update of the binary fonts.
